### PR TITLE
Refactors upsell to use state machine API

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* This changes the scan upsell logic to use a new flexible upsell component.

#### Testing instructions

* In Jetpack, load a site _with_ Scan and observe if it loads.
* Check Scan History as well.
* Load site without Scan and see if upsell displays.

Fixes 1151678672052943-as-1172451663334993.

### CODE REVIEW

**Comment by enejb** [comment]
I tested this and I have some feedback. 

I am not sure that we need the current capabilities api request. The  /scan endpoint (we call it for each site in the side bar already does return the info whether scan is unavailable and for what reason and whether we should be upselling it to the client. There are cased where we wouldn't want to do that. For example rewind engine doesn't support multisite yet so we shouldn't be upselling Jetpack Scan in that case. 

Can we update the Upsell component to use the /scan state data instead?
The rewind endpoint also returns very similar data so we could potentially reuse this component there as well. 

The other things that I noticed is that we flicker the Jetpack Logo. I think it would be useful if we were able to pass in a placeholder component. This way the transition would be much smoother.

**Comment by ChaosExAnima** [comment]
> The other things that I noticed is that we flicker the Jetpack Logo. I think it would be useful if we were able to pass in a placeholder component. This way the transition would be much smoother.

Sure, that's not a problem. What placeholder would be better? We're going to have some sort of flicker as the Jetpack logo happens before hydration, but if we display something completely different that may look better.

**Comment by enejb** [comment]
I think something that represents the page for example Scan History. 

Screen Shot 2020-04-24 at 10 26 56 AM

This gets us closer to what the user would see in the end.

File: client/landing/jetpack-cloud/components/upsell-switch/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# UpsellSwitch
+
+`<UpsellSwitch />` is a component to show an upsell page if a given site's state is unavailable.
```
Comment (rcanepa) [inline client/landing/jetpack-cloud/components/upsell-switch/README.md:3]: I loved this component!

**Comment by ChaosExAnima** [comment]
Merging! @enejb, let's address ineligible sites in another PR? I passed the `reason` as I feel we'll want to recycle assets for upselling instead of having different components for every possible state.

**Pull request merged**